### DISCUSSION TITLE
invalid memory address or nil pointer dereference signal SIGSEGV: segmentation violation

### DISCUSSION THREAD

**Discussion by gowrisankar22** [concourse/concourse#5735] [General]
Hello All,

Nothing really changed but all of the sudden web node went into crashloopback with below error. Not sure what triggered this problem. After delete and recreate the web node fixed the problem.
but still trying to understand root cause of it.

we are using concourse 6.0(k8s)

OCI runtime create failed: container_linux.go:349: starting container process caused "process_linux.go:449: container init caused \"rootfs_linux.go:58: mounting \\\"/var/lib/kubelet/pods/934cb61f-a6e3-451d-98b0-6c8c77c3a60f/volume-subpaths/config-rbac/concourse-web/1\\\" to rootfs \\\"/var/lib/docker/overlay2/97ed4ce3b64a74052f0274ae804447b458c677d30e011ef947949c24e3b7c426/merged\\\" at \\\"/var/lib/docker/overlay2/97ed4ce3b64a74052f0274ae804447b458c677d30e011ef947949c24e3b7c426/merged/config-rbac.yml\\\" caused \\\"no such file or directory\\\"\"": unknown

panic: runtime error: invalid memory address or nil pointer dereference [signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x17bec6a] goroutine 181254874 [running]: github.com/concourse/concourse/atc/exec.(*RetryStep).Succeeded(0xc002db31d0, 0xc0082d8420) /tmp/build/8cf986d9/concourse/atc/exec/retry_step.go:47 +0x2a github.com/concourse/concourse/atc/exec.OnSuccessStep.Succeeded(0x3987c20, 0xc002db31d0, 0x39911a0, 0xc00599a700, 0x1) /tmp/build/8cf986d9/concourse/atc/exec/on_success.go:45 +0x31 github.com/concourse/concourse/atc/exec.OnSuccessStep.Succeeded(0x3987c20, 0xc002db3200, 0x39911a0, 0xc00599a7a0, 0xc006a80ce0) /tmp/build/8cf986d9/concourse/atc/exec/on_success.go:45 +0x5c github.com/concourse/concourse/atc/engine.(*engineBuild).Run(0xc0039f1880, 0x39eed80, 0xc006369e00) /tmp/build/8cf986d9/concourse/atc/engine/engine.go:237 +0xd32 github.com/concourse/concourse/atc/builds.(*Tracker).Track.func1(0xc000db8480, 0x39eed80, 0xc006369800, 0x3a211a0, 0xc003576000) /tmp/build/8cf986d9/concourse/atc/builds/tracker.go:56 +0x320 created by github.com/concourse/concourse/atc/builds.(*Tracker).Track /tmp/build/8cf986d9/concourse/atc/builds/tracker.go:49 +0x2fd

**Comment by vito** [comment]
If you could figure out which pipeline that's coming from, its config would help us figure out the root cause. Looks like the `config-rbac.yml` might be a breadcrumb to follow.

**Comment by gowrisankar22** [comment]
@vito  What's strange is that we could not find any reference[1] [2] to /tmp/build/8cf986d9 anywhere in the DB...

[1]
```
select * from containers where meta_working_directory='/tmp/build/8cf986d9';
0 rows
```
[2]
```
select * from volumes where path='/tmp/build/8cf986d9';
0 rows
```


```
root@concourse-web-68dfbc8487-lpvlw:/# cat  config-rbac.yml
member:
- SaveConfig
- CreateBuild
- DeletePipeline
- OrderPipelines
- HidePipeline
- RenamePipeline
- CreatePipelineBuild
- CreateArtifact
- GetArtifact
- HijackContainer
owner:
- ExposePipeline
- RegisterWorker
- LandWorker
- RetireWorker
- PruneWorker
- HeartbeatWorker
- DeleteWorker
- SetLogLevel
- ReportWorkerContainers
- ReportWorkerVolumes
- ListWorkers
- ListVolumes
- ListContainers
```

**Reply by jamieklassen** [reply]
@gowrisankar22 those file locations must come from golang debug symbols. They refer to the context where the concourse binary was built, which was [this build](https://ci.concourse-ci.org/teams/main/pipelines/release-6.2.x/jobs/build-rc/builds/5#L5ec9bae7:1) on our CI.

**Reply by gowrisankar22** [reply]
@jamieklassen ah ok. But i dont have permission to view https://ci.concourse-ci.org/teams/main/pipelines/release-6.2.x/jobs/build-rc/builds/5#L5ec9bae7:1

**Reply by jamieklassen** [reply]
it's ok, there's nothing to see there, but i exposed it anyway
